### PR TITLE
[BugFix] report error when not-nullable column in external table

### PR DESCRIPTION
**Pull request by dirtysalt**
## What type of PR is this：
- [X] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #https://github.com/StarRocks/starrocks/issues/18348

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

**Comment by DorianZheng** [comment]
Why don't we support non-nullable column direcly?

**Review by Smith-Cruise — DISMISSED** [review]

**Comment by dirtysalt** [comment]
> Why don't we support non-nullable column directly?

It's hard now. All of our code about external table assume that column is nullable. And actually it's true for external table if you use catalog.

**Review by Smith-Cruise — APPROVED** [review]

File: be/src/connector/hive_connector.cpp [hunk 1]
```diff
@@ -47,6 +47,16 @@ DataSourcePtr HiveDataSourceProvider::create_data_source(const TScanRange& scan_
 HiveDataSource::HiveDataSource(const HiveDataSourceProvider* provider, const TScanRange& scan_range)
         : _provider(provider), _scan_range(scan_range.hdfs_scan_range) {}
 
+Status HiveDataSource::_check_all_slots_nullable() {
```
Comment (GavinMar) [inline be/src/connector/hive_connector.cpp:50]: Will it cause some of the previous cases to report errors? Our some test cases have non-nullable columns in iceberg array types.

Comment (dirtysalt) [reply be/src/connector/hive_connector.cpp:50]: Looks like not. be ut has passed. 

but after seocnd thought, we'd better to support non-nullable column.

**Review by GavinMar — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Review by stephen-shelby — APPROVED** [review]

**Review by Youngwb — APPROVED** [review]

**Comment by dirtysalt** [comment]
run starrocks_admit_test

**Comment by dirtysalt** [comment]
run starrocks_admit_test

**Comment by dirtysalt** [comment]
run starrocks_admit_test

**Comment by dirtysalt** [comment]
run starrocks_admit_test

**Pull request merged**